x-pack/plugins/fleet/scripts/create_agents/create_agents.ts

```diff
@@ -0,0 +1,139 @@
+      },
+      user_provided_metadata: {},
+      enrolled_at: new Date().toISOString(),
+      last_checkin: new Date().toISOString(),
+      tags: ['script_create_agents'],
+    }) +
+    '\n'
+  ).repeat(count);
+  const res = await fetch(`${ES_URL}/.fleet-agents/_bulk`, {
+    method: 'post',
+    body,
+    headers: {
+      Authorization: auth,
+      'Content-Type': 'application/x-ndjson',
+    },
+  });
+  const data = await res.json();
+  return data;
+}
+
+async function createSuperUser() {
+  const auth = 'Basic ' + Buffer.from(KIBANA_USERNAME + ':' + KIBANA_PASSWORD).toString('base64');
+  const roleRes = await fetch(`${ES_URL}/_security/role/${ES_SUPERUSER}`, {
+    method: 'post',
+    body: JSON.stringify({
+      indices: [
+        {
+          names: ['.fleet*'],
+          privileges: ['all'],
+          allow_restricted_indices: true,
+        },
+      ],
+    }),
+    headers: {
+      Authorization: auth,
+      'Content-Type': 'application/json',
+    },
+  });
+  const role = await roleRes.json();
+  const userRes = await fetch(`${ES_URL}/_security/user/${ES_SUPERUSER}`, {
+    method: 'post',
+    body: JSON.stringify({
+      password: ES_PASSWORD,
+      roles: ['superuser', ES_SUPERUSER],
+    }),
+    headers: {
+      Authorization: auth,
+      'Content-Type': 'application/json',
+    },
+  });
+  const user = await userRes.json();
+  return { role, user };
+}
+
+async function createAgentPolicy(id: string) {
+  const auth = 'Basic ' + Buffer.from(KIBANA_USERNAME + ':' + KIBANA_PASSWORD).toString('base64');
+  const res = await fetch(`${KIBANA_URL}/api/fleet/agent_policies`, {
+    method: 'post',
+    body: JSON.stringify({
+      id,
+      name: id,
+      namespace: 'default',
+      description: '',
+      monitoring_enabled: ['logs'],
+      data_output_id: 'fleet-default-output',
+      monitoring_output_id: 'fleet-default-output',
+    }),
+    headers: {
+      Authorization: auth,
+      'Content-Type': 'application/json',
+      'kbn-xsrf': 'kibana',
+      'x-elastic-product-origin': 'fleet',
+    },
+  });
+  const data = await res.json();
+  return data;
+}
+
+export async function run() {
+  const logger = new ToolingLog({
+    level: 'info',
+    writeTo: process.stdout,
+  });
+
+  logger.info('Creating agent policy');
+
+  const agentPolicyId = uuid();
+  const agentPolicy = await createAgentPolicy(agentPolicyId);
+  logger.info(`Created agent policy ${agentPolicy.item.id}`);
+
+  logger.info('Creating fleet superuser');
+  const { role, user } = await createSuperUser();
+  logger.info(`Created role ${ES_SUPERUSER}, created: ${role.role.created}`);
+  logger.info(`Created user ${ES_SUPERUSER}, created: ${user.created}`);
+
+  logger.info('Creating agent documents');
+  const count = 50000;
+  const agents = await createAgentDocsBulk(agentPolicyId, count);
+  logger.info(
+    `Created ${agents.items.length} agent docs, took ${agents.took}, errors: ${agents.errors}`
```

Comment:
With this script I managed to test the kibana logic with 200k+ agent docs.
I didn't notice UI slowness even when the async processing took about 60s.

<img width="1132" alt="image" src="https://user-images.githubusercontent.com/90178898/186935250-9b11b6e3-9388-4995-b8a6-cee847258020.png">